### PR TITLE
Refactor/207 port stepper motor profiler cpp

### LINKED ISSUE #207: Port stepperMotorProfiler module to C++
**Describe your use case**
This task will result in a PR that ports the stepperMotorProfiler module to C++.

**Describe alternatives solutions you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by leahkiner**
* **Tickets addressed:** bsk-207
* **Review:** By commit
* **Merge strategy:** Merge (no squash)

## Description
This task ports the `stepperMotorProfiler` module to C++. The new module is named `stepperMotor` and is placed in the `simulation/dynamics` folder.

## Verification
The new C++ module passes the current unit tests.

## Documentation
N/A

## Future work
The module documentation should be written

### CODE REVIEW

File: src/simulation/dynamics/stepperMotor/_UnitTest/test_stepperMotor.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    plt.xlabel('Time (s)', fontsize=14)
+    plt.legend(loc='upper right', prop={'size': 12})
+    plt.grid(True)
+
+    if show_plots:
+        plt.show()
+    plt.close("all")
+
+    # Check to ensure the initial angle converged to the reference angle
+    desiredMotorAngleTrue = initialMotorAngle + (stepsCommanded * stepAngle)
```
Comment (patkenneally) [inline src/simulation/dynamics/stepperMotor/_UnitTest/test_stepperMotor.py:174]: We can remove these testMessages and fail count in favor of np.testing.assert

**Review by patkenneally — APPROVED** [review]
Let's clean up the unit test asserts and then we can merge.

File: src/simulation/dynamics/stepperMotor/_UnitTest/test_stepperMotor.py (hunk 1 above)
Comment (leahkiner) [reply src/simulation/dynamics/stepperMotor/_UnitTest/test_stepperMotor.py:174]: ah I missed this comment, I will do that now.

Comment (leahkiner) [reply src/simulation/dynamics/stepperMotor/_UnitTest/test_stepperMotor.py:174]: Done.

**Pull request merged**